src/does-match.js

```diff
@@ -37,45 +37,108 @@
       return string;
     }
 
-    var replaced = '';
-    for (var i = 0, size = string.length; i < size; i++) {
-      replaced += replaceDiacritics(string[i]);
-    }
+    var replaced = string.split('').map(replaceDiacritics);
 
-    return replaced;
+    return replaced.join('');
   };
 
   /*
     options
   */
-  var validate = {
-    text: function(text) {
-      if (typeof text !== 'string') {
-        throw new Error('`text`: expected string');
-      }
-    },
-    query: function(query) {
-      if (typeof query !== 'string') {
-        throw new Error('`query`: expected string');
-      }
-    },
-    options: function(options) {
-      var minWord = options.minWord;
-      if (minWord !== undefined && typeof minWord !== 'number') {
-        throw new Error('`minWord`: expected number');
-      } else if (minWord < 1 || minWord > 10) {
-        throw new Error('`minWord`: expected number between 1 and 10');
-      }
-      if (options.replaceDiacritics !== undefined && typeof options.replaceDiacritics !== 'boolean') {
-        throw new Error('`replaceDiacritics`: expected boolean');
+  var validation = function() {
+    var proto = {
+      text: function(text) {
+        if (isString(text)) {
+          throw new Error('`text`: expected string');
+        }
+      },
+      query: function(query) {
+        if (isString(query)) {
+          throw new Error('`query`: expected string');
+        }
+      },
+      options: function(options) {
+        var minWord = options.minWord;
+        if (minWord !== undefined && typeof minWord !== 'number') {
+          throw new Error('`minWord`: expected number');
+        } else if (minWord < 1 || minWord > 10) {
+          throw new Error('`minWord`: expected number between 1 and 10');
+        }
+        if (options.replaceDiacritics !== undefined && typeof options.replaceDiacritics !== 'boolean') {
+          throw new Error('`replaceDiacritics`: expected boolean');
+        }
       }
+    };
+
+    function isString(src) {
+      return typeof src !== 'string'
```

Comment:
@ehkasper You should not negate the comparison here, but in the `if`s that call `isString` :+1: